x/programs/rust/examples/multisig/src/lib.rs

```diff
@@ -0,0 +1,495 @@
+    let nays = context.get(StateKeys::Nays(proposal_id)).unwrap().unwrap();
+
+    Ok(_quorum_reached(proposal.voters_len, yeas, nays))
+}
+
+fn _proposal_at(context: &Context<StateKeys>, proposal_id: u64) -> Option<Proposal> {
+    context
+        .get(StateKeys::Proposals(proposal_id))
+        .expect("state corrupt")
+}
+
+fn _last_proposal_id(context: &Context<StateKeys>) -> u64 {
+    context
+        .get(StateKeys::LastProposalId)
+        .expect("state corrupt")
+        .unwrap_or_default()
+}
+
+fn _quorum_reached(max_votes: u64, yeas: u64, nays: u64) -> bool {
+    yeas + nays >= MIN_VOTES && yeas > nays && yeas >= (max_votes + 1) / 2
+}
+
+fn _is_not_duplicate(voters: &[Address]) -> bool {
+    if voters.is_empty() {
+        panic!("there should be at least one voter");
+    }
+
+    for (i, vi) in voters.iter().enumerate() {
+        for (j, vj) in voters[1..].iter().enumerate() {
+            if i != j + 1 && vi == vj {
+                return false;
+            }
+        }
+    }
+
+    true
+}
+
+/// Check if the passed actor is part of the voters and return its index on success
+fn _ensure_voter(actor: &Address, voters: &[Voter]) -> Result<u64, ProposalError> {
+    let mut i = 0;
+    loop {
+        if i < voters.len() {
+            let voter = &voters[i];
+
+            if &voter.address == actor {
+                if voter.voted {
+                    return Err(ProposalError::AlreadyVoted);
+                } else {
+                    break;
+                }
+            }
+
+            i += 1;
+        } else {
+            return Err(ProposalError::NotVoter);
+        }
+    }
+
+    Ok(i as u64)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::ProposalError;
+    use simulator::{Endpoint, Param, Step, StepResponseError, TestContext};
+    use wasmlanche_sdk::{Address, DeferDeserialize};
+
+    const PROGRAM_PATH: &str = env!("PROGRAM_PATH");
+
+    #[test]
+    fn cannot_propose_invalid_with_params() {
+        let mut simulator = simulator::ClientBuilder::new().try_build().unwrap();
+
+        let program_id = simulator
+            .run_step(&Step::create_program(PROGRAM_PATH))
+            .unwrap()
+            .id;
+
+        let test_context = TestContext::from(program_id);
+
+        assert!(matches!(
+            simulator
+                .run_step(&Step {
+                    endpoint: Endpoint::Execute,
+                    method: "propose".to_string(),
+                    max_units: u64::MAX,
+                    params: vec![
+                        test_context.clone().into(),
+                        Vec::new().into(),
+                        program_id.into(),
+                        String::new().into(),
+                        Vec::new().into(),
+                        0u64.into(),
+                    ],
+                })
+                .unwrap()
+                .result
+                .response::<()>(),
+            Err(StepResponseError::ExternalCall(_))
+        ));
```

Comment:
Please use variables

Reply:
Thanks, Created a variable that stored the response and now matches on this one.